### PR TITLE
Second iteration

### PR DESCRIPTION
**Pull request by OlegYankiwskyi**
Changes after comments

### CODE REVIEW

File: Calculator/ViewController.swift [hunk 1]
```diff
@@ -25,182 +50,208 @@ enum Input {
 struct Status {
     var oldResult = 0.0
     var oldOperation: MathOperation = .none
-    var doneOperation = true
+    var isOperationDone = true
     var input: Input = .myValue
```
Comment (ghost) [inline Calculator/ViewController.swift]: rename myValue with something more meaningfull

File: Calculator/ViewController.swift [hunk 2]
```diff
@@ -25,182 +56,206 @@ enum Input {
...
     
     mutating func reset() {
         oldResult = 0.0
         oldOperation = .none
-        doneOperation = true
+        isOperationDone = true
         input = .myValue
```
Comment (vladyslavpetruk-epam) [inline Calculator/ViewController.swift]: newValue?

File: Calculator/ViewController.swift [hunk 3]
```diff
@@ -25,182 +56,206 @@ enum Input {
 struct Status {
     var oldResult = 0.0
     var oldOperation: MathOperation = .none
-    var doneOperation = true
+    var isOperationDone = true
     var input: Input = .myValue
     
     mutating func reset() {
         oldResult = 0.0
         oldOperation = .none
```
Comment (vladyslavpetruk-epam) [inline Calculator/ViewController.swift]: Maybe without `none` case and optional `oldOperation` will be better. Nice case to use optionals!

File: Calculator/ViewController.swift [hunk 4]
```diff
@@ -25,182 +56,206 @@ enum Input {
...
 class ViewController: UIViewController {
-
+    
     @IBOutlet weak var operationLabel: UILabel!
     @IBOutlet weak var resultLabel: UILabel!
     
-    var status = Status()
+    var displayInput: Double? {
+        get {
+            if let input = resultLabel.text , let result = Double(input) {
```
Comment (vladyslavpetruk-epam) [inline Calculator/ViewController.swift]: any ideas how to make it in one line of code? just question :)

File: Calculator/ViewController.swift [hunk 5]
```diff
@@ -25,182 +56,206 @@ enum Input {
...
+        set {
+            if let value = newValue {
+                resultLabel.text = value.parseToString()
+            }
+        }
+    }
     
+    var status = Status()
     
     @IBAction func onButtonDigit(_ sender: UIButton) {
```
Comment (vladyslavpetruk-epam) [inline Calculator/ViewController.swift]: digitButtonTouched

Comment (vladyslavpetruk-epam) [reply Calculator/ViewController.swift]: digitButtonTapped

Comment (vladyslavpetruk-epam) [reply Calculator/ViewController.swift]: digitButtonAction

File: Calculator/ViewController.swift [hunk 6]
```diff
@@ -25,182 +56,206 @@ enum Input {
...
         }
-        
-        if inputValue == "0" {
-            resultLabel.text = "\(sender.tag)"
+
+        switch status.input  {
+        case .result , .oldValue:
+            status.input = .myValue // new Value
+            status.isOperationDone = false
+            displayInput = Double(sender.tag)
```
Comment (vladyslavpetruk-epam) [inline Calculator/ViewController.swift]: make sender.tag more readable

File: Calculator/ViewController.swift [hunk 7]
```diff
@@ -25,182 +56,206 @@ enum Input {
...
+            cleanLabel()
+            status.reset()
+            setOperation()
             return
         }
-        else if inputValue == "-0" {
-            resultLabel.text = "-\(sender.tag)"
+        var result = 0.0
+        
+        guard let operationType = MathOperation(rawValue: sender.tag) else {
```
Comment (vladyslavpetruk-epam) [inline Calculator/ViewController.swift]: combine two guards together

File: Calculator/ViewController.swift [hunk 8]
```diff
@@ -25,182 +56,206 @@ enum Input {
...
+    @IBAction func onUnarOperation(_ sender: UIButton) {
+        guard let newValue = displayInput else {
+            cleanLabel()
+            status.reset()
+            setOperation()
             return
         }
-        else if inputValue == "-0" {
-            resultLabel.text = "-\(sender.tag)"
+        var result = 0.0
```
Comment (vladyslavpetruk-epam) [inline Calculator/ViewController.swift]: I think that `result ` variable is unneeded. You already have `displayInput`.

File: Calculator/ViewController.swift [hunk 9]
```diff
@@ -25,182 +56,206 @@ enum Input {
...
         }
+        displayInput = result
     }
     
-    @IBAction func onButtonOpetation(_ sender: UIButton) {
-        guard let inputValue = resultLabel.text, !inputValue.isEmpty, let newValue = Double(inputValue) else {
-            cleanLable()
+    @IBAction func onBinarOpetation(_ sender: UIButton) {
+        guard let newValue = displayInput else {
+            cleanLabel()
```
Comment (vladyslavpetruk-epam) [inline Calculator/ViewController.swift]: these 3 functions you call from different places. One func with these?

File: Calculator/ViewController.swift [hunk 10]
```diff
@@ -8,12 +8,43 @@
 
 import UIKit
 
+extension Double {
+    func parseToString() -> String {
```
Comment (vladyslavpetruk-epam) [inline Calculator/ViewController.swift]: convertToString?

**Review by vitaliiBoiarskyi — APPROVED** [review]

File: Calculator/ViewController.swift (hunk 4 above)
Comment (OlegYankiwskyi) [reply Calculator/ViewController.swift]: I have one idea : return Double(resultLabel.text!) ?? nil 
But this idea uses force unwrap and I think that It is bad idea due to force unwrap

**Pull request merged**